--- PEAR-1.9.4/usr/share/pear/pearcmd.php~	2014-05-14 19:21:49.000000000 +0300
+++ PEAR-1.9.4/usr/share/pear/pearcmd.php	2014-05-14 19:21:53.064418441 +0300
@@ -45,6 +45,16 @@
 
 $pear_package_version = "1.10.15";
 
+// before trying anything deep, ensure base extensions are present
+// this avoids debugging problems with bogus error messages, which really are
+// caused by missing extension
+foreach (array('pcre', 'zlib', 'xml') as $ext) {
+	if (!extension_loaded($ext)) {
+		error_log("pear requires PHP extension '$ext' for ".PHP_VERSION);
+		exit(1);
+	}
+}
+
 require_once 'PEAR.php';
 require_once 'PEAR/Frontend.php';
 require_once 'PEAR/Config.php';
