--- ./usr/share/pear/PEAR/Command/Package.php	2005-09-18 16:42:13.000000000 +0300
+++ /usr/share/pear/PEAR/Command/Package.php	2005-09-18 16:57:39.000000000 +0300
@@ -877,6 +877,11 @@
         $info['package'] = $pf->getPackage();
         $info['version'] = $pf->getVersion();
         $info['release_license'] = $pf->getLicense();
+        $info['release_state'] = $pf->getState();
+		list($info['class'], $info['subclass']) = explode('_', $info['package'], 2);
+		if (empty($info['subclass'])) {
+			$info['subclass'] = '%{nil}';
+		}
         if ($pf->getDeps()) {
             if ($pf->getPackagexmlVersion() == '1.0') {
                 $requires = $conflicts = array();
--- .//usr/share/pear/PEAR/Command/Package.php	2005-09-28 23:18:13.000000000 +0300
+++ /usr/share/pear/PEAR/Command/Package.php	2005-09-28 23:16:37.000000000 +0300
@@ -908,7 +908,7 @@
                         $chan = &$reg->getChannel($dep['channel']);
                         $package = strtoupper($chan->getAlias()) . '::' . $dep['name'];
                     } else {
-                        $package = 'PEAR::' . $dep['name'];
+                        $package = 'php-pear-' . $dep['name'];
                     }
                     $trans = array(
                         '>' => '>',
@@ -933,10 +933,14 @@
                     }
                 }
                 if (count($requires)) {
-                    $info['extra_headers'] .= 'Requires: ' . implode(', ', $requires) . "\n";
+					foreach ($requires as $dep) {
+						$info['extra_headers'] .= "Requires:\t" . $dep . "\n";
+					}
                 }
                 if (count($conflicts)) {
-                    $info['extra_headers'] .= 'Conflicts: ' . implode(', ', $conflicts) . "\n";
+					foreach ($conflicts as $dep) {
+						$info['extra_headers'] .= "Conflicts:\t" . $dep . "\n";
+					}
                 }
             } else {
                 $info['package2xml'] = '2'; // tell the spec to use package2.xml
@@ -951,7 +955,7 @@
                             $chan = &$reg->getChannel($dep['channel']);
                             $package = strtoupper($chan->getAlias()) . '::' . $dep['name'];
                         } else {
-                            $package = 'PEAR::' . $dep['name'];
+                            $package = 'php-pear-' . $dep['name'];
                         }
                         if (!isset($dep['min']) && !isset($dep['max']) &&
                               !isset($dep['exclude'])) {
@@ -996,15 +1000,19 @@
                         $tar->popErrorHandling();
                         if ($a === null || PEAR::isError($a)) {
                             // this doesn't have a package.xml version 1.0
-                            $requires[] = 'PEAR::PEAR >= ' .
+                            $requires[] = 'php-pear-PEAR >= ' .
                                 $deps['required']['pearinstaller']['min'] . "\n";
                         }
                     }
                     if (count($requires)) {
-                        $info['extra_headers'] .= 'Requires: ' . implode(', ', $requires) . "\n";
+						foreach ($requires as $dep) {
+							$info['extra_headers'] .= "Requires:\t" . $dep . "\n";
+						}
                     }
                     if (count($conflicts)) {
-                        $info['extra_headers'] .= 'Conflicts: ' . implode(', ', $conflicts) . "\n";
+						foreach ($conflicts as $dep) {
+							$info['extra_headers'] .= "Conflicts:\t" . $dep . "\n";
+						}
                     }
                 }
             }
--- .//usr/share/pear/PEAR/Command/Package.php	2005-09-28 23:25:11.000000000 +0300
+++ /usr/share/pear/PEAR/Command/Package.php	2005-09-28 23:39:21.000000000 +0300
@@ -814,7 +814,7 @@
         }
 
         $info['extra_headers'] = '';
-        $info['doc_files'] = '';
+        $info['doc_files'] = array();
         $info['files'] = '';
         $info['package2xml'] = '';
         $info['rpm_package'] = sprintf($rpm_pkgname_format, $pf->getPackage());
@@ -825,7 +825,7 @@
             }
             $name = preg_replace('![/:\\\\]!', '/', $name);
             if ($attr['role'] == 'doc') {
-                $info['doc_files'] .= " $name";
+                $info['doc_files'][] = $name;
             // Map role to the rpm vars
             } else {
                 $c_prefix = '%{_libdir}/php/pear';
@@ -863,6 +863,16 @@
                 $info['files'] .= "$prefix/$name\n";
             }
         }
+
+		$ndocs = count($info['doc_files']);
+		if ($ndocs > 1) {
+			$info['doc_files'] = '%doc docs/%{_pearname}/{' . join(',', $info['doc_files']) . '}';
+		} elseif ($ndocs > 0) {
+			$info['doc_files'] = '%doc docs/%{_pearname}/' . join(',', $info['doc_files']);
+		} else {
+			$info['doc_files'] = '';
+		}
+
         if ($srcfiles > 0) {
             require_once 'OS/Guess.php';
             $os = new OS_Guess;
@@ -894,10 +904,13 @@
 			$info['subclass'] = '%{nil}';
 		}
         if ($pf->getDeps()) {
+			$info['optional'] = '';
+
             if ($pf->getPackagexmlVersion() == '1.0') {
                 $requires = $conflicts = array();
                 foreach ($pf->getDeps() as $dep) {
                     if (isset($dep['optional']) && $dep['optional'] == 'yes') {
+						$info['optional'] .= "'pear(". str_replace('_', '/', $dep['name']) . ".*)' ";
                         continue;
                     }
                     if ($dep['type'] != 'pkg') {
@@ -1020,6 +1033,7 @@
 
         // remove the trailing newline
         $info['extra_headers'] = trim($info['extra_headers']);
+        $info['optional'] = trim($info['optional']);
         if (function_exists('file_get_contents')) {
             fclose($fp);
             $spec_contents = preg_replace('/@([a-z0-9_-]+)@/e', '$info["\1"]',
--- .//usr/share/pear/PEAR/Command/Package.php	2005-09-28 23:55:36.000000000 +0300
+++ /usr/share/pear/PEAR/Command/Package.php	2005-09-28 23:55:06.000000000 +0300
@@ -816,6 +816,8 @@
         $info['extra_headers'] = '';
         $info['doc_files'] = array();
         $info['files'] = '';
+        $info['test_files'] = '';
+        $info['data_files'] = '';
         $info['package2xml'] = '';
         $info['rpm_package'] = sprintf($rpm_pkgname_format, $pf->getPackage());
         $srcfiles = 0;
@@ -828,7 +830,7 @@
                 $info['doc_files'][] = $name;
             // Map role to the rpm vars
             } else {
-                $c_prefix = '%{_libdir}/php/pear';
+                $c_prefix = '%{php_pear_dir}';
                 switch ($attr['role']) {
                     case 'php':
                         $prefix = $c_prefix;
@@ -860,7 +862,15 @@
                     break;
                 }
                 $name = str_replace('\\', '/', $name);
-                $info['files'] .= "$prefix/$name\n";
+				if ($attr['role'] == 'test') {
+					$info['test_files'] .= "$prefix/$name\n";
+
+				} elseif ($attr['role'] == 'data') {
+					$info['data_files'] .= "$prefix/$name\n";
+
+				} else {
+					$info['files'] .= "$prefix/$name\n";
+				}
             }
         }
 
@@ -872,6 +882,7 @@
 		} else {
 			$info['doc_files'] = '';
 		}
+		$info['have_tests'] = (strlen($info['test_files']) > 0) + 0;
 
         if ($srcfiles > 0) {
             require_once 'OS/Guess.php';
@@ -1034,6 +1045,8 @@
         // remove the trailing newline
         $info['extra_headers'] = trim($info['extra_headers']);
         $info['optional'] = trim($info['optional']);
+		$info['have_optional_deps'] = (strlen($info['optional']) > 0) + 0;
+
         if (function_exists('file_get_contents')) {
             fclose($fp);
             $spec_contents = preg_replace('/@([a-z0-9_-]+)@/e', '$info["\1"]',
--- .//usr/share/pear/PEAR/Command/Package.php	2005-09-29 00:26:51.000000000 +0300
+++ /usr/share/pear/PEAR/Command/Package.php	2005-09-29 00:26:38.000000000 +0300
@@ -1024,7 +1024,7 @@
                         $tar->popErrorHandling();
                         if ($a === null || PEAR::isError($a)) {
                             // this doesn't have a package.xml version 1.0
-                            $requires[] = 'php-pear-PEAR >= ' .
+                            $requires[] = 'php-pear-PEAR >= 1:' .
                                 $deps['required']['pearinstaller']['min'] . "\n";
                         }
                     }
--- .//usr/share/pear/PEAR/Command/Package.php	2005-09-29 00:58:11.000000000 +0300
+++ /usr/share/pear/PEAR/Command/Package.php	2005-09-29 01:06:46.000000000 +0300
@@ -862,6 +862,9 @@
                     break;
                 }
                 $name = str_replace('\\', '/', $name);
+				if ($attr['baseinstalldir']) {
+					$name = $attr['baseinstalldir'] . '/' . $name;
+				}
 				if ($attr['role'] == 'test') {
 					$info['test_files'] .= "$prefix/$name\n";
 
