--- PEAR-1.4.8/usr/share/pear/PEAR/Command/Package.php~	2006-03-06 03:09:16.000000000 +0200
+++ PEAR-1.4.8/usr/share/pear/PEAR/Command/Package.php	2006-03-06 03:11:58.000000000 +0200
@@ -819,8 +819,10 @@
         if (isset($options['rpm-pkgname'])) {
             $rpm_pkgname_format = $options['rpm-pkgname'];
         } else {
-            if ($pf->getChannel() == 'pear.php.net' || $pf->getChannel() == 'pecl.php.net') {
-                $alias = 'PEAR';
+            if ($pf->getChannel() == 'pear.php.net') {
+                $alias = 'php-pear';
+            } elseif ($pf->getChannel() == 'pecl.php.net') {
+                $alias = 'php-pecl';
             } else {
                 $chan = &$reg->getChannel($pf->getChannel());
                 if (PEAR::isError($chan)) {
@@ -830,7 +832,7 @@
                 $alias = strtoupper($alias);
                 $info['possible_channel'] = $pf->getChannel() . '/';
             }
-            $rpm_pkgname_format = $alias . '::%s';
+            $rpm_pkgname_format = $alias . '-%s';
         }
 
         $info['extra_headers'] = '';
