diff -burN PEAR-1.4.5.orig/usr/share/pear/PEAR/Command/Package.php PEAR-1.4.5/usr/share/pear/PEAR/Command/Package.php
--- PEAR-1.4.5.orig/usr/share/pear/PEAR/Command/Package.php	2005-12-20 08:14:58.485437880 +0100
+++ PEAR-1.4.5/usr/share/pear/PEAR/Command/Package.php	2005-12-20 08:15:37.777464584 +0100
@@ -1184,7 +1184,7 @@
                 fread($fp, filesize($spec_template)));
             fclose($fp);
         }
-        $spec_file = "$info[rpm_package]-$info[version].spec";
+        $spec_file = "$info[rpm_package].spec";
         $wp = fopen($spec_file, "wb");
         if (!$wp) {
             return $this->raiseError("could not write RPM spec file $spec_file: $php_errormsg");
