--- ./usr/bin/pear~	2005-09-30 17:11:26.000000000 +0300
+++ ./usr/bin/pear	2005-09-30 17:13:14.000000000 +0300
@@ -4,10 +4,12 @@
 if test "x$PHP_PEAR_PHP_BIN" != "x"; then
   PHP="$PHP_PEAR_PHP_BIN"
 else
-  if test "/usr/bin/php" = '@'php_bin'@'; then
-    PHP=php 
-  else
+  if [ -x "/usr/bin/php" ]; then
     PHP="/usr/bin/php"
+  elif [ -x "/usr/bin/php4" ]; then
+    PHP="/usr/bin/php4"
+  else
+    PHP="php"
   fi
 fi
 
