--- /usr/share/pear/PEAR.php~	2007-07-04 01:22:17.000000000 +0300
+++ /usr/share/pear/PEAR.php	2007-07-04 16:25:12.627146755 +0300
@@ -566,10 +566,10 @@
             $ec = 'PEAR_Error';
         }
         if ($skipmsg) {
-            $a = &new $ec($code, $mode, $options, $userinfo);
+            $a = new $ec($code, $mode, $options, $userinfo);
             return $a;
         } else {
-            $a = &new $ec($message, $code, $mode, $options, $userinfo);
+            $a = new $ec($message, $code, $mode, $options, $userinfo);
             return $a;
         }
     }
