--- ./pearcmd.php	2005-11-28 08:32:28.000000000 +0200
+++ /usr/share/pear/pearcmd.php	2005-11-08 23:59:21.000000000 +0200
@@ -26,17 +26,6 @@
     define('PEAR_RUNTYPE', 'pear');
 }
 define('PEAR_IGNORE_BACKTRACE', 1);
-if (!function_exists('file_get_contents')) {
-    function file_get_contents($filename)
-    {
-        $fp = fopen($filename, 'rb');
-        $ret = '';
-        while (!feof($fp)) {
-            $ret .= fread($fp, 8092);;
-        }
-        return $ret;
-    }
-}
 /**
  * @nodep Gtk
  */
